controller/src/test/java/io/pravega/controller/store/stream/TableHelperTest.java

```diff
@@ -789,7 +790,7 @@ public void scaleInputValidityTest() {
     public void truncationTest() {
```

Comment:
the problem is we try to create a state in the test by doing multiple scales and truncations. and during each state change, we perform certain assertions and then finally some overall assertions.. 
we can make this code more readable by breaking it into multiple methods.. but it will still be a single test..

Reply:
It is not the first PR in which I see this happening. The problem is that this test case is growing without bounds, and that's not good. We need to come up with a way to test all these scenarios without requiring an uber test case that tests everything related to truncation. In fact, I'd argue that it would be much cleaner if we created the state for each scenario being tested individually for the sake of separating concerns.

If you want to do it separately, then fine, I don't think a refactoring of this test case is in scope for this PR, but please create an issue to address this.

Reply:
#2766